### PR TITLE
Assets pallet: Don't allow `set_min_balance` when sufficient

### PR DESCRIPTION
**Pull request by Szegoo**
As mentioned [here](https://github.com/paritytech/substrate/pull/13486#issuecomment-1451664289) we shouldn't allow asset owners to decrease the `min_balance` when the asset is marked as sufficient.

### CODE REVIEW

**Comment by Szegoo** [comment]
cc @jsidorenko @bkchr

File: frame/assets/src/lib.rs [hunk 1]
```diff
@@ -1540,10 +1540,15 @@ pub mod pallet {
 			ensure!(origin == details.owner, Error::<T, I>::NoPermission);
 
 			let old_min_balance = details.min_balance;
-			// Ensure that either the new min_balance is less than old min_balance or there aren't
-			// any accounts holding the asset.
+			let is_sufficient = details.is_sufficient;
+			// Ensure that either the new min_balance is less than old
+			// min_balance or there aren't any accounts holding the asset.
+			//
+			// If the asset is marked as sufficient we will only allow the owner
```
Comment (IkerAlus) [inline frame/assets/src/lib.rs:1547]: what is the rationale behind allowing this when there is no accounts holding the asset? I would say that the sufficiency check should apply in any case.

Comment (Szegoo) [reply frame/assets/src/lib.rs:1547]: You are right, it doesn't make sense.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2474329

**Review by jsidorenko — APPROVED** [review]

**Comment by jsidorenko** [comment]
bot bench $ pallet dev pallet_assets

File: frame/assets/src/lib.rs [hunk 2]
```diff
@@ -1540,8 +1540,13 @@ pub mod pallet {
 			ensure!(origin == details.owner, Error::<T, I>::NoPermission);
 
 			let old_min_balance = details.min_balance;
-			// Ensure that either the new min_balance is less than old min_balance or there aren't
-			// any accounts holding the asset.
+			let is_sufficient = details.is_sufficient;
+			// If the asset is marked as sufficient it won't be allowed to
+			// change the min_balance.
+			ensure!(!is_sufficient, Error::<T, I>::NoPermission);
```
Comment (bkchr) [inline frame/assets/src/lib.rs:1546]:
```suggestion
			// If the asset is marked as sufficient it won't be allowed to
			// change the min_balance.
			ensure!(!details.is_sufficient, Error::<T, I>::NoPermission);
```

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
@Szegoo please merge master.

**Comment by Szegoo** [comment]
@bkchr Merged, but I think we need to rerun the benchmark here.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2476232

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2476226

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2476225

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2476283

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2476277

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2476278

**Comment by Szegoo** [comment]
@bkchr Merged. I think we may have to rerun the benchmark.

**Comment by jsidorenko** [comment]
bot bench $ pallet dev pallet_assets

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-release-analysis-v0-9-40/2468/1

**Pull request merged**